reducers/views/channel.js

```diff
@@ -115,11 +115,58 @@ function keepChannelIdAsUnread(state = null, action) {
     }
 }
 
+function channelPostsStatus(state = {}, action) {
+    switch (action.type) {
+    case ActionTypes.CHANNEL_POSTS_STATUS: {
+        const channelId = action.data.channelId;
+        if ((action.data).hasOwnProperty('atEnd')) {
+            return {
+                ...state,
+                [channelId]: {
+                    ...state[channelId],
+                    atEnd: action.data.atEnd,
+                },
+            };
+        }
+        return {
+            ...state,
+            [channelId]: {
+                ...state[channelId],
+                atStart: action.data.atStart,
+            },
+        };
+    }
+    default:
+        return state;
+    }
+}
+
+function channelSyncStatus(state = {}, action) {
+    switch (action.type) {
+    case GeneralTypes.WEBSOCKET_FAILURE || GeneralTypes.CLOSED: {
```

Comment:
Regarding my earlier comment, maybe we could just change this to clear the synced status of every channel when the websocket disconnects like
```javascript
const nextState = {};
for (const channelId of Object.keys(state)) {
    nextState[channelId] = false;
}
return nextState;
```
That way, we don't need to pass the channel IDs down from the websocket disconnection code

Reply:
I need to clear the flags for all the channels. 

Once a socket disconnects and connects only channel in view calls syncPosts and makes api calls. Every other channel syncs upon viewing the channel(Calls syncPost on load based on sync flag)